vent/archive/controls.py

```diff
@@ -1,166 +1,166 @@
-        
-    def __hold_peep(self):
-        self.expir.off()
-        #self.inspir.on()
-        self.logger.calculate_last_tv()
-        self.__print_breath(self.logger.tv)
-        self.__inc_state()
-        
-    def __print_status(self):
-        # change this to grabbing last observations from logger
-        print("STATE: %2d pressure_0: %4.1f pressure_1: %4.1f Temp: %4.1f flow: %4.0f"%(self._state,self.jp.get_pressure(1),self.jp.get_pressure(1),self.jp.get_temperature(),self.jp.get_flow()))#,end="\r")
-
-    def __print_breath(self,tv):
-        return
-        # TODO -------------------------------------------------
-        #print("\n Last breath: %2.1f seconds, %4.0f mL tidal volume"%(0,0))
-
-class DataLogger:
-# A class for keeping track of prior sensor values.
-# These are used by controllers and to sound alarms.
-    def __init__(self, jp):
-        self.jp = jp
-        self._track_len = 50    # Type int: How long to retain sensor data. Could vary this later per sensor. 
-        self.pres1_track = []
-        self.pres2_track = []
-        self.o2_track = []
-        self.flow_track = []
-        self.temp_track = []
-        self.humid_track = []
-        self.tv     = []
-
-    def update(self):
-    # Takes in current sensor readings object; updates tracking log for all sensors.
-        self.pres1_track = self.update_single_sensor(self.pres1_track, self.jp.get_pressure(0), self._track_len)
-        self.pres2_track = self.update_single_sensor(self.pres2_track, self.jp.get_pressure(1), self._track_len)
-        self.o2_track = self.update_single_sensor(self.o2_track, self.jp.get_o2(), self._track_len)
-        self.flow_track = self.update_single_sensor(self.flow_track, self.jp.get_flow(), self._track_len)
-        self.temp_track = self.update_single_sensor(self.temp_track, self.jp.get_temperature(), self._track_len)
-        self.humid_track = self.update_single_sensor(self.humid_track, self.jp.get_humidity(), self._track_len)
-
-    def calculate_last_tv(self):
-        # integrate flow over the last breath
-        # TODO -------------------------------------------------
-        self.tv = -1
-        return self.tv
-    
-    def reset(self):
-        self.pres1_track = []
-        self.pres2_track = []
-        self.o2_track = []
-        self.flow_track = []
-        self.temp_track = []
-        self.humid_track = []
-
-    def update_single_sensor(self, track, val, track_len):
-        # Adds newest sensor reading to end of track.
-        # Only store most recent [track_len] values.
-        
-        # Check if list is already at max track length:
-        if(len(track) >= track_len):
-            # Remove first element
-            track.pop(0)
-        # Add current sensor reading to track:
-        track.append(val)  
-        return track 
+# library imports:
+import time
+import gpiozero
+from sensors import JuliePlease
+#import RPi.GPIO as GPIO   # Import the GPIO library.
+
+class ControlSettings:

```

Comment:
I'm going to close this PR, clean up the history and open a new one. A bunch of files got caught in here because I botched a commit and then reverted with another commit, but everything that got caught up in that is still listed in files changed.

Reply:
See #45 

This should also help (somewhat) with your comment about 28 files being touched.